docs(platform): correct how assets become repositories, and answer the adoption questions (#622)
convisoappsec/conviso-docs: docs/platform/repositories-and-branches.md

  risk of the code you actually ship.

:::note
Assets created before the repository model become repositories when your scans report a repository
address **and** a branch; anything left over is combined with the help of the Conviso team. See
[How Your Assets Become Repositories](#how-your-assets-become-repositories).
:::

## Do I Need to Do Anything?

There is **nothing for you to turn on**. The repository model does not have a setting in your
account, and no migration runs on your side. If **Repositories** is not listed under
**Inventory > Assets** yet, ask Conviso Support where your account stands.

What is worth doing is making sure your pipelines report what the model needs, and that the branch
the platform will measure is the branch you actually ship.

### Where you need to act

| Situation | What to do |
| --- | --- |

> | Your pipeline does not report a **repository address and a branch** | Act now. Without both, nothing is consolidated and the branch is ignored — with no error to tell you |

Reviewer: bloqueante: "nothing is consolidated" não é garantido. Sem a branch o resultado depende do caminho: pode não consolidar, ou consolidar adotando uma branch que ninguém declarou — e aí o repositório passa a medir a branch errada, que é pior do que não consolidar.

```suggestion
| Your pipeline does not report a **repository address and a branch** | Act now. Without both, the consolidation either does not happen or lands on a branch you did not declare — with no error to tell you |
```

> | The same codebase is registered under two addresses — HTTPS and SSH, or a browser URL | Standardize on one address now: two addresses become two separate repositories |

Reviewer: sugestão: HTTPS e SSH do mesmo servidor normalmente resolvem para o mesmo repositório; o que separa de verdade é o servidor cujos dois endereços não se parecem — Azure DevOps é o caso, e é exatamente o que a página já diz em "Registering a Repository". Do jeito que está, manda todo cliente de GitHub/GitLab "agir agora" sem ter o que fazer.

```suggestion
| The same codebase is registered under two addresses that do not look alike — the HTTPS and the SSH form on Azure DevOps, say, or a browser URL | Standardize on one address now: addresses that differ in host or path become two separate repositories |
```

| More than one asset carries the same repository address | Act now. With two or more candidates nothing is consolidated on its own, and the case has to go to the Conviso team |
| Your production branch is not the one marked as default | Adjust it before the first scan — see [Setting the Branch That Represents Production](#setting-the-branch-that-represents-production) |
| You report the risk score upward | Note the current numbers first: the score starts following the default branch alone |

### Does it break anything?

**What does not change:**

* The **id of the asset**. Ids of assets that became branches keep resolving permanently — in the
  interface, in the API and in your integrations.
* Your **findings**, with their status, comments, assignees and history.
* Your **defect-tracker links**, **bookmarks** and deep links.
* Your **pipelines and scanner integrations**, which keep reporting without being edited.

**What does change:**

* The **risk score** reflects the default branch alone, and the open-vulnerability counters no
  longer add up every branch.
* **Dashboards, exports and the Security Gate** follow the default branch. See

>   [What Each Screen Shows](#what-each-screen-shows).

Reviewer: sugestão: contradiz a tabela que o próprio bullet linka. Em "What Each Screen Shows", a lista de vulnerabilidades, seus contadores e o CSV dela mostram todas as branches. "Exports" no plural e "os contadores" sem qualificar valem para o repositório, não para a lista.

```suggestion
* The **risk score** reflects the default branch alone, and the repository's open-vulnerability
  counters — the list column and the Details card — no longer add up every branch.
* **Dashboards, the Repositories export and the Security Gate** follow the default branch; the
  vulnerability list and the export made from it still show every branch. See
  [What Each Screen Shows](#what-each-screen-shows).
```

* Rows that used to be **separate assets collapse into a single repository row**.
* **Business Impact**, **Data Classification** and **Attack Surface** of the assets that were
  absorbed are dropped — review them on the repository afterwards.

## Before You Start

### What your pipeline has to report

Two values, and **both are required**: the **repository address** and the **branch**.

:::caution
A scan that reports the repository address without a branch — or a branch without the address — is
**ignored silently**. There is no error and no warning: the scan finishes, the findings land where
they always did, and nothing is consolidated.

> :::

Reviewer: bloqueante: mesma raiz do comentário na linha 46, e aqui é o parágrafo que o leitor usa para decidir se precisa mexer no pipeline. "Ignored silently / nothing is consolidated" tranquiliza na direção errada: quem manda o endereço sem a branch pode acabar com um repositório já criado medindo uma branch que não é a dele.

```suggestion
:::caution
A scan that reports the repository address without a branch — or a branch without the address —
does not consolidate the way you expect, and nothing tells you so: there is no error and no
warning. Depending on the path, nothing is consolidated at all, or the repository ends up
measuring a branch you never declared.
:::
```

There is **no minimum version** enforced by the platform. What matters is capability, not version:
whatever reports your scans has to be able to send both values. If you use a Conviso task or
plugin in your pipeline, check that it is on a version that offers repository and branch fields.

The [GitHub Actions integration](../integrations/github-actions.md#repository-and-branch) fills
both in from the workflow context by default, and is a good reference for what the equivalent

> fields look like elsewhere. For scanner integrations, each one asks for the repository and most

Reviewer: nit: a âncora existe, mas a seção é a do **GitHub Sync Task** (importação de scanners externos). Quem roda a action de AST scan segue o link e não acha o caso dele.

```suggestion
The [GitHub Sync Task action](../integrations/github-actions.md#repository-and-branch) fills both
in from the workflow context by default, and is a good reference for what the equivalent fields
look like elsewhere. For scanner integrations, each one asks for the repository and most
```

ask for a branch — see
[Importing projects from a scanner integration](#importing-projects-from-a-scanner-integration).

### The access you need

**No new permission is introduced.** The permissions you already use on assets cover branches:

| What you want to do | What you need |
| --- | --- |
| See a repository's branches | The same access that lets you see the asset |
| **Set as default**, **Rename branch**, **Delete branch**, **New Branch** | **Update** permission on Asset — and the asset must not be archived |
| Register a repository | **Create** permission on Asset |

> | Combine repositories, or split one back into separate assets | Conviso Support — this is not an action available in your account |

Reviewer: sugestão: falta a permissão que mais causa "por que não consolidou": consolidar mexe num asset que a chamada não criou, então exige permissão sobre esse asset. Sem ela o scan cria uma entrada separada e segue, sem erro. E vale a ressalva de não empurrar Delete para chave de pipeline como se fosse recomendação.

```suggestion
**No new permission is introduced.** The permissions you already use on assets cover branches:

| What you want to do | What you need |
| --- | --- |
| See a repository's branches | The same access that lets you see the asset |
| **Set as default**, **Rename branch**, **Delete branch**, **New Branch** | **Update** permission on Asset — and the asset must not be archived |
| Register a repository | **Create** permission on Asset |
| Have a scan consolidate an asset you already had | Permission on that asset: **Update**, and **Delete** when an existing repository absorbs it |
| Combine repositories, or split one back into separate assets | Conviso Support — this is not an action available in your account |

The last two are worth checking against the credentials your pipeline uses: without that
permission the scan does not consolidate, it creates a separate entry — with no error. If you
would rather not grant it to a pipeline key, ask Conviso Support to consolidate instead.
```

required.

:::caution
Changing the default branch **recalculates the repository's risk score**, because the repository's
risk always follows the default branch. The change raises a notification of its own, so whoever
follows the repository sees why the number moved — default-branch changes made by Conviso while

> consolidating an account are not notified.

Reviewer: dúvida: a frase é verdadeira, mas é a doc pública assumindo que a Conviso muda a branch medida sem avisar. É decisão de produto/comunicação, não minha — se ficar, eu emendaria com onde ver isso depois, já que a página diz logo abaixo que o histórico de mudanças de default branch se pede ao Suporte.

Accounts that used the platform before the repository model have assets representing individual
branches. They become repositories two ways.

### When a scan reports a repository and a branch

An asset is turned into a repository on its own when a scan — or a call to the API that creates or
updates it — reports a **repository address together with a branch**, and the match leaves no room
for doubt. All of the following have to hold:

* Both values are reported. **The branch is required**: an address without a branch, or a branch

>   without an address, does nothing.

Reviewer: bloqueante: terceira ocorrência da mesma afirmação (ver linha 78). "does nothing" é a parte que não se sustenta.

```suggestion
* Both values are reported. **Always report the branch**: without it the platform cannot tell
  which branch the findings belong to, and the consolidation either does not happen or adopts a
  branch nobody declared.
```

* **Exactly one** asset carries that repository address. With two or more candidates nothing is
  consolidated, and the case goes to the Conviso team.
* The asset is **not archived**, and it is not a cloud, domain or API asset.
* Whoever makes the call has **Update** permission on the asset.

When they all hold, one of two things happens:

* If no repository is registered at that address yet, the asset **becomes the repository** for it,
  keeping its id, its name, its findings and its history. The branch that was reported becomes the
  repository's default branch.
* If a repository is already registered at that address, the asset is **absorbed into it** as the
  branch that was reported, and stops appearing as a row of its own. Absorbing also requires
  **Delete** permission on the asset being absorbed.

:::caution
The branch is what makes this safe, and that is why it is required. Consolidating without one would
file the scan under a placeholder default branch that nothing ever corrects — and because the
repository's risk score follows its default branch, a critical finding reported on `staging` would
leave the repository reading **zero**.

> :::

Reviewer: sugestão: continuação do ponto acima. O condicional ("would file the scan under a placeholder") promete que isso não acontece; é justamente o risco que existe quando a branch não vem. No indicativo o aviso fica mais útil e continua verdadeiro.

```suggestion
:::caution
The branch is what keeps this safe. Without one the platform has nothing to file the scan under,
and the repository can end up with a default branch nobody declared — and because the risk score
follows its default branch, a critical finding reported on `staging` leaves the repository reading
**zero**.
:::
```

:::note
Everything outside those conditions stays exactly as it is: assets nothing scans, scans that report
no address, scans that report an address without a branch, and addresses claimed by more than one
asset. None of them are consolidated automatically — they are handled with the Conviso team, below.

> :::

Reviewer: sugestão: "Everything outside those conditions stays exactly as it is" (e o "All of the following have to hold" na linha 456) prometem uma exaustividade que a página não consegue sustentar sem virar especificação. Um caso que falta e o cliente encontra: se o repositório já tem uma branch com o nome que o scan reportou, o asset antigo não é absorvido e fica ao lado.

```suggestion
:::note
Outside those conditions nothing is consolidated on its own: assets nothing scans, scans that
report no address, scans that report an address without a branch, and addresses claimed by more
than one asset. There are narrower cases too — if the repository already has a branch with the
name your scan reported, the asset is not folded into it and stays beside it. All of them are
handled with the Conviso team, below.
:::
```

### With the Conviso team

Whatever is left — assets no pipeline reaches, duplicates created by hand, the same address claimed
by several assets, groupings you want reviewed — is combined by the **Conviso team**. Combining is
always a reviewed action: the platform can suggest which assets look like the same repository, but

> nothing is merged without someone deciding it. Contact Conviso Support if you want your account

Reviewer: sugestão: "the platform can suggest which assets look like the same repository" descreve ferramenta interna que o cliente não vê nem aciona. O que interessa a ele é a segunda metade: ninguém funde nada sem decisão humana.

```suggestion
Whatever is left — assets no pipeline reaches, duplicates created by hand, the same address claimed
by several assets, groupings you want reviewed — is combined by the **Conviso team**. Combining is
always a reviewed action: nothing is merged without someone deciding it. Contact Conviso Support if
you want your account
```
